newIDE/app/src/MainFrame/index.js

```diff
@@ -635,23 +662,24 @@ class MainFrame extends React.Component<Props, State> {
     );
     if (!answer) return;
 
-    this.setState(
-      {
+    setState(state =>
+      ({
+        ...state,
         editorTabs: closeExternalLayoutTabs(
-          this.state.editorTabs,
+          state.editorTabs,
           externalLayout
         ),
-      },
-      () => {
+      }),
+      (state) => {
```

Comment:
Well, we'd need to always pass `state` in the callback, otherwise, it will be called with the previous value of `state`. This is why I'm not really super happy about `useStateWithCallback` :/

Reply:
Sure, but in most cases you don't need state in the callbacks. And because forceUpdate does NOT need state, you can remove it entirely from most callbacks :)